scripts/RelayHubGasEstimation.ts

```diff
@@ -54,374 +63,429 @@ function logTitle(title: string) {
+  relayHub = await deployRelayHub(penalizer.address);
+  await setupRelayHub(relayHub);
+
+  await fundedAccount.sendTransaction({
+    to: owner.address,
+    value: ethers.utils.parseEther('1'),
+  });
 
-  function printGasAnalysis({
-    cumulativeGasUsed,
-    detailedEstimation,
-    internalDestinationCallCost,
-    internalTokenCallCost,
-  }: {
-    internalDestinationCallCost: number;
-    internalTokenCallCost: number;
-    detailedEstimation: number;
-    cumulativeGasUsed: number;
-  }) {
-    console.log(
-      'The destination contract call estimate is: ',
-      internalDestinationCallCost
-    );
-    console.log('The token gas estimate is: ', internalTokenCallCost);
-    console.log(
-      'X = ',
-      internalDestinationCallCost + internalTokenCallCost
-    );
+  const factory = (await createSmartWalletFactory(
+    smartWalletTemplate,
+    false,
+    owner
+  )) as SmartWalletFactory;
 
-    console.log(`Detailed estimation: ${detailedEstimation}`);
-    console.log(`Cumulative Gas Used: ${cumulativeGasUsed}\n`);
-  }
+  forwarder = (await createSupportedSmartWallet({
+    relayHub: relayHubSigner.address,
+    factory,
+    owner,
+    sender: relayHubSigner,
+    logGas: true,
+  })) as SmartWallet;
 
-  const assertRelayedTransaction = async (
-    forwarderInitialBalance: BigNumber,
-    relayWorkerInitialBalance: BigNumber,
-    balanceToTransfer: BigNumber,
-    fees?: BigNumber
-  ) => {
-    const forwarderFinalBalance = await token.balanceOf(
-      forwarder.address
-    );
-    const relayWorkerFinalBalance = await token.balanceOf(
-      relayWorker.address
-    );
+  await token.mint('1000', forwarder.address);
 
-    expect(
-      forwarderInitialBalance.eq(
-        forwarderFinalBalance
-          .add(balanceToTransfer)
-          .add(BigNumber.from(fees ?? 0))
-      )
-    ).to.equal(true, 'SW Payment did not occur');
-    expect(
-      relayWorkerFinalBalance.eq(
-        relayWorkerInitialBalance.add(fees ?? balanceToTransfer)
-      )
-    ).to.equal(true, 'Worker did not receive payment');
+  relayRequest = {
+    request: {
+      relayHub: relayHub.address,
+      to: recipient.address,
+      data: '0xdeadbeef',
+      from: owner.address,
+      nonce: (await forwarder.nonce()).toString(),
+      value: '0',
+      gas: '3000000',
+      tokenContract: token.address,
+      tokenAmount: '1',
+      tokenGas: '50000',
+      validUntilTime: '0',
+    },
+    relayData: {
+      gasPrice,
+      feesReceiver: relayWorker.address,
+      callForwarder: forwarder.address,
+      callVerifier: verifier.address,
+    },
   };
 
-  const assertNonce = async (nonceBefore: BigNumber) => {
-    const nonceAfter = await forwarder.nonce();
-    expect(nonceBefore.add(1).toNumber()).to.equal(
-      nonceAfter.toNumber(),
-      'Incorrect nonce after execution'
-    );
+  return {
+    forwarder,
+    recipient,
+    relayHub,
+    owner,
+    relayRequest,
+    factory,
+    token,
+    verifier
```

Comment:
if we return those objects here, why do we need to have them also defined globally? I'd remove the global definition to have them returned here only

Reply:
Done.